### PR TITLE
[OSDOCS-4594]: Adding vSphere CPMS config details

### PR DESCRIPTION
**Pull request by jeana-redhat**
Version(s):
4.12+

Issue:
[OSDOCS-4594](https://issues.redhat.com//browse/OSDOCS-4594)

Link to docs preview:
[Sample YAML for configuring VMware vSphere clusters](https://53107--docspreview.netlify.app/openshift-enterprise/latest/machine_management/control_plane_machine_management/cpmso-configuration.html#cpmso-sample-yaml-vsphere_cpmso-configuration)

QE review:
- [ ] QE has approved this change.

Additional information:
Also fixes incorrect metadata on the other YAML reference modules

### CODE REVIEW

File: modules/cpmso-yaml-provider-spec-vsphere.adoc [hunk 1]
```diff
@@ -1,22 +1,53 @@
...
+  snapshot: ""
+  template: <vm_template_name> <8>
+  userDataSecret:
+    name: master-user-data <9>
+  workspace: 
+    datacenter: <vcenter_datacenter_name> <10>
+    datastore: <vcenter_datastore_name> <11>
+    folder: <path_to_vcenter_vm_folder> <12>
+    resourcePool: <vsphere_resource_pool> <13>
+    server: <vcenter_server_ip> <14>
```
Comment (sunzhaohua2) [inline modules/cpmso-yaml-provider-spec-vsphere.adoc:38]: @jeana-redhat  missing indentation

File: modules/cpmso-yaml-provider-spec-vsphere.adoc [hunk 2]
```diff
@@ -1,22 +1,53 @@
...
 ----
+<1> Specifies the secret name for the cluster. Do not change this value.
+<2> Specifies the VM size for the control plane machines.
```
Comment (JoelSpeed) [inline modules/cpmso-yaml-provider-spec-vsphere.adoc:41]: VM disk size?

File: modules/cpmso-yaml-provider-spec-vsphere.adoc [hunk 3]
```diff
@@ -1,22 +1,53 @@
...
+<3> Specifies the cloud provider platform type. Do not change this value. 
+<4> Specifies the memory allocated for the control plane machines.
+<5> Specifies the network on which the control plane is deployed.
+<6> Specifies the number of CPUs allocated for the control plane machines.
+<7> Specifies the number of cores for the control plane machine CPUs.
```
Comment (JoelSpeed) [inline modules/cpmso-yaml-provider-spec-vsphere.adoc:46]: cores per control plane machine CPU?

Wondering if the current wording is ambiguous as to whether this is a total or per CPU

File: modules/cpmso-yaml-provider-spec-vsphere.adoc (hunk 1 above)
Comment (jeana-redhat) [reply modules/cpmso-yaml-provider-spec-vsphere.adoc:38]: Thanks! Fixed :)

**Comment by jeana-redhat** [comment]
Feedback addressed, thanks @sunzhaohua2 & @JoelSpeed! 

I am going to submit for editorial review since this was straightforward and the US is heading into a holiday weekend. Will not merge until you've had a chance to look, so likely Monday when I am back in.

/label peer-review-needed

**Comment by JoelSpeed** [comment]
LGTM from SME perspective

File: modules/cpmso-yaml-sample-cr.adoc [hunk 4]
```diff
@@ -54,11 +54,15 @@ $ oc get -o jsonpath='{.status.infrastructureName}{"\n"}' infrastructure cluster
 +
 [IMPORTANT]
 ====
-Before you activate the Operator, you must ensure that the `ControlPlaneMachineSet` CR configuration is correct for your cluster requirements.
-//For more information about activating the Control Plane Machine Set Operator, see "Getting started with the Control Plane Machine Set Operator".
+Before you activate the Operator, you must ensure that the `ControlPlaneMachineSet` CR configuration is correct for your cluster requirements. For more information about activating the Control Plane Machine Set Operator, see "Getting started with the Control Plane Machine Set Operator".
```
Comment (abhatt-rh) [inline modules/cpmso-yaml-sample-cr.adoc:57]: In the preview, the left hand navigation lists "Getting started". Consider updating it to "Getting started with the Control Plane Machine Set Operator" for ease of search and to comply with other sibling topics that mention the full Operator name.

**Review by abhatt-rh — COMMENTED** [review]
Hi @jeana-redhat
Nice and neat- LGTM! I left a suggestion for your consideration, though. 

/remove-label peer-review-in-progress
/remove-label peer-review-needed
/label peer-review-done

**Pull request merged**